.github/workflows/e3sm-gh-clang-format.yml

```diff
@@ -0,0 +1,73 @@
+name: "clang format"
+
+# if .{cpp,hpp} files are touched in a PR, lint them!
+
+on:
+  pull_request:
+    branches: ["master"]
+    paths:
+      - 'components/eamxx/**/*.cpp'
+      - 'components/eamxx/**/*.hpp'
+      - '.github/workflows/e3sm-gh-clang-format.yml'
```

Comment:
Should we call the file `eamxx-gh-clang-format.yml`, to avoid giving false expectations?